Add documentation for `ClimateMachine` settings (#1326)
CliMA/ClimateMachine.jl: docs/src/GettingStarted/RunningClimateMachine.md

## Input and output

The `ClimateMachine` provides the [`ArtifactWrappers`](@ref
ArtifactWrappers-docs) module to assist a driver in sourcing input data for a
simulation, but any mechanism may be used.

Output takes the form of various [groups of diagnostic variables](@ref

> Diagnostics-vars) that are written to NetCDF files at specified intervals by
> the `ClimateMachine` when configured to do so by a driver.

Reviewer: ```suggestion
Diagnostics-vars) that are written to NetCDF files at user-specified intervals
in the `ClimateMachine` driver `Callbacks`.
```
Maybe include a ref to the callbacks / driver?

Reviewer: The suggestion is just a suggestion. I'm not sure if what I wrote is more transparent, but I think something along those lines would be clear/helpful

The `ClimateMachine` can also output conservative and auxiliary state variables
to VTK files at specified intervals.

Whether or not output is generated, and if so, at what interval, is a
`ClimateMachine` _setting_.

## `ClimateMachine` settings

Some aspects of the `ClimateMachine`'s behavior can be controlled via its
_settings_ such as use of the GPU, diagnostics output and frequency,
checkpointing/restarting, etc. There are 3 ways in which these settings can be
changed:

1. [Command line arguments](@ref ClimateMachine-clargs) have the highest
   precedence, but it is possible for a driver to disable parsing of command
   line arguments. In such a case, only the next two ways can be used to change
   settings.

> 2. [Environment variables](@ref ClimateMachine-envvars) are considered after
>    command line arguments.

Reviewer: ```suggestion
2. [Environment variables](@ref ClimateMachine-envvars) take precedence after
   command line arguments.
```

3. [Programmatic settings](@ref ClimateMachine-kwargs) have the lowest

>    precedence.

Reviewer: ❤️ 
I can see using this as a common reference to new users. Looks great.

Reviewer 2: @kpamnany ENV variables have the lowest precedence

Author: Doh! You're right...

### [Command line arguments](@id ClimateMachine-clargs)

If a driver configures the `ClimateMachine` to parse command line

> arguments (most do), you can query the list of arguments understood,

Reviewer 3: ```suggestion
arguments by setting `parse_clargs = true` in `ClimateMachine.init()`, you can query the list of arguments understood,
```
I think there should be some indication for developers on how to turn this flag on.
